CAIPs/caip-x.md

```diff
@@ -0,0 +1,110 @@
+### Chain Agnostic Improvement Proposal (CAIP) for Wallet-to-DApp Communication Standard
```

Comment:
If we end up merging this one instead of Pedro's version in PR #282, just snatch the frontmatter from that version and the number, since it would close issue #294 , after all!